src/cosalib/qemuvariants.py

```diff
@@ -0,0 +1,220 @@
+        "image_suffix": "vhd",
+        "platform": "azure",
+    },
+    "openstack": {
+        "image_format": "qcow2",
+        "platform": "openstack",
+    },
+    "gcp": {
+        "image_format": "raw",
+        "platform": "gcp",
+        "image_suffix": "tar.gz",
+        "tar_member": True,
+    },
+    "vmware_vmdk": {
+        "image_format": "vmdk",
+        "image_suffix": "vmdk",
+        "platform": "vmware",
+        "convert_options":  {
+            '-o': 'adapter_type=lsilogic,subformat=streamOptimized,compat6'
+        }
+    }
+}
+
+
+class ImageError(Exception):
+    """
+    Base error for build issues.
+    """
+    pass
+
+
+def get_qemu_variant(variant, parser, kwargs={}):
+    """
+    Helper function get get the QemuVariantImage Build Obj
+    """
+    log.debug(f"returning QemuVariantImage for {variant}")
+    return QemuVariantImage(
+        parser.buildroot,
+        parser.build,
+        variant=variant,
+        **kwargs)
+
+
+class QemuVariantImage(_Build):
+    def __init__(self, *args, **kwargs):
+        """
+        This takes all the regular _BuildClass arguments. In kwargs, the
+        additional arguments are used:
+            image_format: standard qemu types
+            convert_options: optional qemu arguments
+            platform: the name of the image.
+            platform_image_name: in case you want to use a different name
+
+        Alternatively, you can provide "variant=<variant>" and defaults will be
+        used.
+        """
+        self._image_name = None
+        variant = kwargs.get("variant")
+        if variant is not None:
+            kwargs.update(VARIANTS[variant])
+
+        self.image_format = kwargs.get("image_format")
+        self.image_suffix = kwargs.get("image_suffix", self.image_format)
+        self.convert_options = kwargs.get("convert_options", {})
+        self.platform = kwargs.get("platform")
+        self.force = kwargs.get("force", False)
+        self.tar_member = kwargs.get("tar_member", False)
+        self.tar_member_name = kwargs.get("tar_memeber_name", "disk.raw")
+
+        # this is used in case the image has a different disk
+        # name than the platform
+        self.platform_image_name = kwargs.get(
+            "platform_image_name", self.platform)
+
+        _Build.__init__(self, *args, **kwargs)
+
+    @property
+    def image_qemu(self):
+        qimage = os.path.join(
+            self.build_dir,
+            self.meta.get('images', {}).get('qemu', {}).get('path', None)
+        )
+        if not qimage:
+            raise ImageError("qemu image has not be built yet")
+        elif not os.path.exists(qimage):
+            raise ImageError(f"{qimage} does not exist")
+        return qimage
+
+    @property
+    def image_name(self):
+        return f'{self.image_name_base}.{self.image_suffix}'
+
+    @image_name.setter
+    def image_name(self, val):
+        self._image_name = val
+
+    @property
+    def tmp_image(self):
```

Comment:
This bit feels weird to me - a `QemuVariantImage` has-a `tmp_image`?  This feels like a more functional thing - there's a `transform_image_platform(src, dest)` method that takes a `QemuVariantImage` and gives you a new one.